pkg/image/build.go

```diff
@@ -51,7 +51,7 @@ func Build(cfg *config.Config, dir, imageName string, secrets []string, noCache,
 			return fmt.Errorf("Failed to generate weights manifest: %w", err)
 		}
 		cachedManifest, err := weights.LoadManifest(weightsManifestPath)
-		changed := err != nil && weightsManifest.Equal(cachedManifest)
+		changed := err != nil || !weightsManifest.Equal(cachedManifest)
```

Comment:
Let's just return err if not nil. That's easier to tell 😄 
Then make:

```go
changed := !weightsManifest.Equal(cachedManifest)
```

Reply:
Well, if there are no cached weights (i.e the first time this is executed), there will be `err`. Which I think is OK. I would have checked `cachedManifest == nil || !weights.Equal(cached)`. What do you think?

Reply:
Alternatively we can put a check for `nil` inside `Equal` method, after all why not compare against `nil`.

Reply:
+1 for `cachedManifest == nil || !weights.Equal(cached)` as the check. That closely tracks my understanding for how the feature behaves.

Reply:
Use os.IsNotExist() to tell if there is no cached weights (i.e. cachedManifest == nil)